shared/src/question-generators/math/crt.ts

```diff
@@ -0,0 +1,166 @@
+  },
+}
+
+function generateCRTQuestion(lang: Language, path: string, random: Random, numCongruences: number) {
+  const congruences: { a: number; n: number }[] = []
+  const text: string[] = []
+
+  // generate system of congruences
+  for (let i = 0; i < numCongruences; i++) {
+    let a = random.int(1, 20)
+    let n: number
+
+    // find new modulus that is coprime to preceding moduli
+    do {
+      n = random.int(2, 20)
+    } while (congruences.some(({ n: precedingModulus }) => !areCoprime(n, precedingModulus)))
+
+    a = ((a % n) + n) % n // a should be within the range [0, n-1]
+    congruences.push({ a, n })
+    text.push(`x \\equiv ${a} \\pmod{ ${n} }`)
+  }
+
+  const crtValue = solveCRT(congruences)
+  const commonModulus = congruences.reduce((acc, { n }) => acc * n, 1)
+
+  const question: FreeTextQuestion = {
+    type: "FreeTextQuestion",
+    name: CRT.name(lang),
+    path: path,
+    text: t(translations, lang, "crtQuestion", { text: text.join(", \\\\") }),
+    prompt: `$x \\equiv $`,
+    bottomText: t(translations, lang, "crtBottomText"),
+    feedback: getCRTFeedbackFunction(lang, crtValue, commonModulus),
+    checkFormat: getCRTCheckFormatFunction(lang),
+  }
+
+  return { question, testing: { crtValue, commonModulus } }
+}
+
+function getCRTCheckFormatFunction(lang: Language): FreeTextFormatFunction {
+  return ({ text }) => {
+    if (text.trim() === "") {
+      return { valid: false, message: t(translations, lang, "feedbackIncomplete") }
+    }
+
+    // ensure format "$value (mod $modulus)" and inform user
+    const pattern = /^(\d+)\s*\(\s*mod\s*(\d+)\s*\)$/i
+    const match = text.trim().match(pattern)
+    if (!match) {
+      return { valid: false, message: t(translations, lang, "feedbackIncomplete") }
+    }
+
+    return { valid: true }
+  }
+}
+
+function getCRTFeedbackFunction(
+  lang: Language,
+  crtValue: number,
+  commonModulus: number,
+): FreeTextFeedbackFunction {
+  return ({ text }) => {
+    // match "y (mod z)" (optional whitespaces) and capture y and z
+    const pattern = /^(\d+)\s*\(\s*mod\s*(\d+)\s*\)$/i
+    const match = text.trim().match(pattern)
+
+    // dismiss incorrectly formatted answers
+    if (!match) {
+      return { correct: false, feedbackText: t(translations, lang, "feedbackInvalid") }
+    }
+
+    // y: userValue and z: userModulus
+    let userValue = parseInt(match[1], 10)
+    const userModulus = parseInt(match[2], 10)
+
+    // normalize to range [0, commonModulus]
+    userValue = ((userValue % commonModulus) + commonModulus) % commonModulus
+    crtValue = ((crtValue % commonModulus) + commonModulus) % commonModulus
+
+    const isCorrect = userModulus === commonModulus && userValue === crtValue
+
+    return {
+      correct: isCorrect,
+      feedbackText: isCorrect
+        ? t(translations, lang, "feedbackCorrect")
+        : t(translations, lang, "feedbackIncorrect", {
+            correctAnswer: `${crtValue} (mod ${commonModulus})`,
+          }),
```

Comment:
Case the answer is correct, I would only return `{correct: isCorrect}`
Otherwise: 
```
return {
      correct: isCorrect,
      correctAnswer: isCorrect ? "" : `${crtValue} (mod ${commonModulus})`
}
```
Something like this, because we already have a block for the correctAnswer and feedbackText: 
tcs.uni-frankfurt.de/algo-learn/en/plnf/input/2/2b7t26o

This is also for some other questions